generate_cs_arch_globals.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python
+
+"""Generates global Capstone variable declarations"""
+
+from __future__ import print_function
+
+import os
+import sys
+import time
+
+
+ARCH_INFO = [
+    ("CAPSTONE_HAS_ARM", "CS_ARCH_ARM", "ARM",
+     ["CS_MODE_LITTLE_ENDIAN", "CS_MODE_ARM", "CS_MODE_V8",
+      "CS_MODE_MCLASS", "CS_MODE_THUMB", "CS_MODE_BIG_ENDIAN"]),
+    ("CAPSTONE_HAS_ARM64", "CS_ARCH_ARM64", "AArch64",
+     ["CS_MODE_LITTLE_ENDIAN", "CS_MODE_ARM", "CS_MODE_BIG_ENDIAN"]),
+    ("CAPSTONE_HAS_MIPS", "CS_ARCH_MIPS", "Mips",
+     ["CS_MODE_LITTLE_ENDIAN", "CS_MODE_32", "CS_MODE_64", "CS_MODE_MICRO",
+      "CS_MODE_MIPS32R6", "CS_MODE_MIPSGP64", "CS_MODE_BIG_ENDIAN"]),
+    ("CAPSTONE_HAS_X86", "CS_ARCH_X86", "X86",
+     ["CS_MODE_LITTLE_ENDIAN", "CS_MODE_32", "CS_MODE_64", "CS_MODE_16"]),
+    ("CAPSTONE_HAS_POWERPC", "CS_ARCH_PPC", "PPC",
+     ["CS_MODE_LITTLE_ENDIAN", "CS_MODE_32", "CS_MODE_64",
+      "CS_MODE_BIG_ENDIAN"]),
+    ("CAPSTONE_HAS_SPARC", "CS_ARCH_SPARC", "Sparc",
+     ["CS_MODE_BIG_ENDIAN", "CS_MODE_V9"]),
+    ("CAPSTONE_HAS_SYSZ", "CS_ARCH_SYSZ", "SystemZ",
+     ["CS_MODE_BIG_ENDIAN"]),
+    ("CAPSTONE_HAS_XCORE", "CS_ARCH_XCORE", "XCore",
+     ["CS_MODE_BIG_ENDIAN"]),
+]
+
+GLOBALS_FILE = 'cs_arch_globals.c'
+
+
+def interpolate_format(format_, arch_info, modes_formatter=None):
+    """Print the interpolated output of each arch"""
+    if modes_formatter is None:
+        modes_formatter = lambda x: ""
+
+    for cs_has, cs_arch, pretty_name, allowed_modes in arch_info:
+        print(format_.format(
+            cs_has=cs_has, cs_arch=cs_arch, pretty_name=pretty_name,
+            allowed_modes=modes_formatter(allowed_modes)).strip())
+
+
+ARCH_INIT_FMT = """
+#ifdef {cs_has}
+\t{pretty_name}_global_init,
+#else
+\tNULL,
+#endif
+"""
+
+ARCH_OPTION_FMT = """
+#ifdef {cs_has}
+\t{pretty_name}_option,
+#else
+\tNULL,
+#endif
+"""
+
+ARCH_DESTROY_FMT = """
+#ifdef {cs_has}
+\t{pretty_name}_destroy,
+#else
+\tNULL,
+#endif
+"""
+
+ALLOWED_MODES_FMT = """
+#ifdef {cs_has}
+\t{allowed_modes},
+#else
+\t0,
+#endif
+"""
+
+ALL_ARCH_FMT = """
+#ifdef {cs_has}
+\t(1 << {cs_arch})
+#else
+\t0
+#endif
+"""
+
+HEADER = """
+/* Capstone Disassembly Engine */
+/* By Nguyen Anh Quynh <aquynh@gmail.com>, {year} */
```

Comment:
no, this is for auto-generated file, not created by me ;-)

to credit, please add your name into this new file (generate_cs_arch_globals.py) if you want that.